blosc2/blosc2_ext.pyx

```diff
@@ -498,11 +540,34 @@ def get_blocksize():
     """
     return blosc1_get_blocksize()
 
+cdef _check_cparams(blosc2_cparams *cparams):
+    if cparams.nthreads > 1:
+        if BLOSC2_USER_REGISTERED_CODECS_START <= cparams.compcode <= BLOSC2_USER_REGISTERED_CODECS_STOP:
+            raise ValueError("Cannot use multi-threading with user defined codecs")
+        elif any([BLOSC2_USER_REGISTERED_FILTERS_START <= filter <= BLOSC2_USER_REGISTERED_FILTERS_STOP
+                  for filter in cparams.filters]):
+            raise ValueError("Cannot use multi-threading with user defined filters")
+        elif cparams.prefilter != NULL:
+            raise ValueError("`nthreads` must be 1 when a prefilter is set")
```

Comment:
"... is set" -> "... is set.  In the future this limitation could be dropped tough."